### ISSUE TITLE
tab order bug

### ISSUE THREAD

**Issue by kevindherman** [proudcity/wp-proudcity#2330]
## Source
Marin Accessibility

* [TICKET](https://proudcity.zendesk.com/agent/tickets/6538)

## Describe the bug
The Digital Accessibility division of the County’s IST department contacted me saying that some of our page templates on the Public Works website (https://publicworks.marincounty.org/) do not function correctly for accessibility requirements. The issue is the “tab order” and how it seems to get lost on some pages after getting past the breadcrumb line. They showed it to me in action on several of the project webpages (https://publicworks.marincounty.org/bolinas-road-slide-repair/) as well as some of the “sub-landing page” pages we have such as this one (https://publicworks.marincounty.org/printing-services/) . They used a tool called “ANDI” to demonstrate the gaps in tab order. I’m thinking it might have something to do with any page that has the left-side navigation column? Anyway, can you please look into it and let me know what can be done? Happy to hop on a video call to show you what I’m talking about, if that would be helpful?

**CM**

Our issue is that we have "hidden" menu items that show up higher in the page hierarchy so they show up first in the `tabindex` before the menu directly on screen is in the HTML hierarchy. To add to this, we don't have good `:focus` styles on the menu at all.

- The main widget wrapper for this is here: https://github.com/proudcity/wp-proud-core/blob/master/modules/proud-widget/widgets/submenu/submenu-widget.class.php
- It calls `Core\ProudMenu`: https://github.com/proudcity/wp-proud-core/blob/3aa78fd61fabce8460b12d4f0e2ed03a18ac65b1/modules/proud-menu/proud-menu.php#L146
- templates for the markup are located: https://github.com/proudcity/wp-proud-core/tree/master/modules/proud-menu/templates

Somewhere in [this js function](https://github.com/proudcity/wp-proud-core/blob/3aa78fd61fabce8460b12d4f0e2ed03a18ac65b1/modules/proud-menu/assets/js/proud-menu.js#L38) we should be able to loop through all the `level-*` classes and add `tabindex=-1` to them while skipping the active level as defined by `level- + levelTo` so that it stays in the `tabindex` order.

- [x] fix tabindex
- [x] add the `:focus` and `:active` states in styles for the menu

## QA

Notes:

* [x] check recovers sites specifically

Links:

*

**Comment by curtismchale** [comment]
@kevindherman @lukefretwell after looking at this, the work was in the digging. I think I have a workable solution, though it's not going to address every edge case.

We should be able to add `tabindex=-1` to all the links in the menu while skipping the currently active menu. Then we'd do the same thing when the Back button is used, setting the tab index to start where the back button just navigated to.

I want to highlight that this is a theoretical solution. It should work, but I wouldn't be surprised if there are other edge cases I'm currently missing in my investigation.

**Comment by kevindherman** [comment]
@curtismchale @lukefretwell sounds like it's worth testing as an intermediary step. @lukefretwell what do you think?

**Comment by lukefretwell** [comment]
@curtismchale @kevindherman worth a try 🙏

**Comment by curtismchale** [comment]
This was deployed to `beta` and `marindpw` for testing.

**Comment by curtismchale** [comment]
Released

Build: https://github.com/proudcity/proud-recipes/releases/tag/2023.09.19.1208
composer: https://github.com/proudcity/wp-proudcity/releases/tag/2023.09.19.1208
